stl/inc/span

```diff
@@ -628,14 +639,14 @@ public:
     // [span.iterators] Iterator support
     _NODISCARD constexpr iterator begin() const noexcept {
 #if _ITERATOR_DEBUG_LEVEL >= 1
-        return {_Mydata, _Mydata, _Mydata + this->size()};
+        return {_Mydata, _Mydata, _Mydata + size()};
 #else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
         return {_Mydata};
 #endif // _ITERATOR_DEBUG_LEVEL
     }
 
     _NODISCARD constexpr iterator end() const noexcept {
-        const auto _End = _Mydata + this->size();
+        const auto _End = _Mydata + size();

```

Comment:
Had an interesting idea which is in the neighborhood of this refactoring - whenever we need the size, we used to call `size()` in the base class, to handle the empty optimization. But what if, for static extents, the base class provided a `static constexpr size_t _Mysize = _Extent`? Then `span` could just refer to `_Mysize`, avoiding function calls in debug mode (and inlining in release mode).

The "cost" would be the somewhat unusual complexity of `_Mysize` being either static or non-static, but that isn't *too* weird. (At most we might get constant comparison warnings?)

Reply:
Loving it. Pushed an implementation